be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -104,13 +106,27 @@ Status HorizontalGeneralTabletWriter::flush_segment_writer(SegmentPB* segment) {
 VerticalGeneralTabletWriter::VerticalGeneralTabletWriter(TabletManager* tablet_mgr, int64_t tablet_id,
                                                          std::shared_ptr<const TabletSchema> schema, int64_t txn_id,
                                                          uint32_t max_rows_per_segment)
-        : TabletWriter(tablet_mgr, tablet_id, std::move(schema), txn_id), _max_rows_per_segment(max_rows_per_segment) {}
+        : TabletWriter(tablet_mgr, tablet_id, std::move(schema), txn_id),
+          _max_rows_per_segment(max_rows_per_segment),
+          _async_write_segment(config::lake_enable_async_segment_writer) {}
 
 VerticalGeneralTabletWriter::~VerticalGeneralTabletWriter() = default;
 
 // To developers: Do NOT perform any I/O in this method, because this method may be invoked
 // in a bthread.
 Status VerticalGeneralTabletWriter::open() {
+    if (_async_write_segment) {
+        // CAUTION: we reuse delta writer's memory table flush pool here
+        auto flush_pool = StorageEngine::instance()->memtable_flush_executor()->get_thread_pool();
+        if (UNLIKELY(flush_pool == nullptr)) {
+            return Status::InternalError("Get memory table flush pool failed");
+        }
+        // Use CONCURRENT mode to ensure segments can finalize in parallel
+        _segment_writer_finalize_token = flush_pool->new_token(ThreadPool::ExecutionMode::CONCURRENT);
```

Comment:
Instead of create a threadpool token here, can we pass in a threadpool token externally to TabletWriter? By doing so, we can remove the dependency of `memtable_flush_executor()` and the global configuration `config::lake_enable_async_segment_writer` from the TabletWriter.

Reply:
By doing so, we can support async write in the HorizontalTabletWriter also.